sys/riotboot/hdr.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_riotboot_hdr
+ * @{
+ *
+ * @file
+ * @brief       RIOT header helpers and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stddef.h>
+#include <stdio.h>
+
+#ifdef RIOT_VERSION
+#include "log.h"
+#else
+#define LOG_INFO(...) printf(__VA_ARGS__)
+#endif
+
+#include "riotboot/hdr.h"
+#include "checksum/fletcher32.h"
+#include "byteorder.h"
+
+#if __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__
+#   error "This code is implementented in a way that it will only work for little-endian systems!"
+#endif
+
+void riotboot_hdr_print(const riotboot_hdr_t *riotboot_hdr)
+{
+    printf("Image magic_number: 0x%08x\n", (unsigned)riotboot_hdr->magic_number);
+    printf("Image Version: 0x%08x\n", (unsigned)riotboot_hdr->version);
+    printf("Image start address: 0x%08x\n", (unsigned)riotboot_hdr->start_addr);
+    printf("Header chksum: 0x%08x\n", (unsigned)riotboot_hdr->chksum);
+    printf("\n");
+}
+
+int riotboot_hdr_validate(const riotboot_hdr_t *riotboot_hdr)
+{
+    if (riotboot_hdr->magic_number != RIOTBOOT_MAGIC) {
+        LOG_INFO("%s: riotboot_hdr magic number invalid\n", __func__);
```

Comment:
But since it's an error case, it should be `LOG_ERROR`. The same goes line 59 in this file.

Reply:
In this and the other case it would imply to redefine all the `LOG_*` macros according to the debug level (in other words, re-implement the whole logging module after the `#ifdef`), which is not the goal in this use case. The goal here is to just inform if something went wrong even if there are failures out of the RIOT context (e.g. a external tool which might want to validate a header).

Another option would be to put `#ifdef`s before each all to match the context on which this file is compiled.